src/Sets/Zonotope.jl

```diff
@@ -371,19 +371,179 @@ nonlinear systems with uncertain parameters using conservative linearization.
 In Proc. of the 47th IEEE Conference on Decision and Control.*
 """
 function split(Z::Zonotope, j::Int)
-    @assert 1 <= j <= ngens(Z) "cannot split a zonotope with $(ngens(Z)) generators along index $j"
     c, G = Z.center, Z.generators
-    Gj = G[:, j]
-    Gj_half = Gj / 2
-
-    c₁ = c - Gj_half
-    c₂ = c + Gj_half
-
-    G₁ = copy(G)
-    G₁[:, j] = Gj_half
-    G₂ = copy(G₁)
 
+    c₁ = similar(c)
+    G₁ = similar(G)
     Z₁ = Zonotope(c₁, G₁)
+
+    c₂ = similar(c)
+    G₂ = similar(G)
```

Comment:
i think that was to share code (with / without sarrays). also, the `split!` method is interesting on its own, there are cases when it's useful to pass preallocated arrays in advance.

Reply:
But at the moment I think there is no shared code.

Reply:
with the new commit there's shared code. some comments

- `similar` is used because that produces an MArray (mutable statically sized array) from an SArray
- `copyto!` works regardless of the array type
- in the end we need to convert to SArray again so that is the particular method (this is optional, but conventional)

Reply:
I still do not see the purpose of doing `similar` followed by `copyto!`. Why not do `copy` immediately? I do not see a use case for passing dummy zonotopes instead of creating the zonotopes inside the method.

Reply:
The idea is that the in-place version `split!` receives preallocated arrays (zonotope center / generators matrices) and overrides whatever is such arrays with new values. 

> Why not do copy immediately? Does that not work for MArray?

Copying an SArray creates an SArray, but we want to modify values so we need an intermediate MArray to work with. See also [here](https://juliaarrays.github.io/StaticArrays.jl/latest/pages/api/#Mutable-arrays:-MVector,-MMatrix-and-MArray-1).

Reply:
About output preallocation, using the pattern `split` / `split!` is comparable to https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Pre-allocating-outputs-1.

Reply:
Ah, I see, makes sense now.